### ISSUE TITLE
Crash in addBreadcrumb: NSInvalidArgumentException: Invalid number value (infinite) in JSON write

### ISSUE THREAD

**Issue by ronyfadel** [getsentry/sentry-cocoa#3936]
### Platform

macOS

### Environment

Production

### Installed

Swift Package Manager

### Version

7.31.5

### Steps to Reproduce

If I try to add a breadcrumb with a number set to +inf (e.g. CMTime.positiveInfinity.seconds), Sentry crashes my app.

code to repro:

```
public extension SentrySDK {
  static func addBreadcrumb(_ message: String, level: SentryLevel = .info, category: String? = nil, data: [String: Any]? = nil) {
    let crumb = Breadcrumb()
    crumb.message = message
    crumb.level = level
    if let category {
      crumb.category = category
    }
    crumb.data = data
    self.addBreadcrumb(crumb: crumb)
  }
}

SentrySDK.addBreadcrumb("test", data: ["foo": CMTime.positiveInfinity])
```

crumb right before:
```
{

crumb: {
error: Invalid JSON data: Invalid character,
json_data: {"category":"default","data":{"language":"it","duration":inf,"translate_to":"fa"},"level":"info","message":"ac_subtitle_start","timestamp":"2024-05-05T01:07:36.116Z"}
},
language: it
}
```

### Expected Result

App doesn't crash.

### Actual Result

app crashes with.

**Comment by ronyfadel** [comment]
cc @philipphofmann

**Comment by philipphofmann** [comment]
@ronyfadel, this could be fixed with https://github.com/getsentry/sentry-cocoa/pull/3458 released in [8.17.1](https://github.com/getsentry/sentry-cocoa/blob/main/CHANGELOG.md#8171). Can you try to update to this version and see if the problem persists?

**Comment by ronyfadel** [comment]
Confirmed working
